plugins/woocommerce-admin/client/products/fills/product-form-fills.tsx

```diff
@@ -1,52 +1,152 @@
+	INVENTORY_SECTION_ADVANCED_ID,
+	PLUGIN_ID,
+} from './constants';
 
+const Tabs = () => {
+	const { values: product } = useFormContext< Product >();
+	const tabPropData = useMemo(
+		() => ( {
+			general: {
+				name: 'general',
+				title: __( 'General', 'woocommerce' ),
+			},
+			pricing: {
+				name: 'pricing',
+				title: __( 'Pricing', 'woocommerce' ),
+				disabled: !! product?.variations?.length,
+			},
+			inventory: {
+				name: 'inventory',
+				title: __( 'Inventory', 'woocommerce' ),
+				disabled: !! product?.variations?.length,
+			},
+			shipping: {
+				name: 'shipping',
+				title: __( 'Shipping', 'woocommerce' ),
+				disabled: !! product?.variations?.length,
+			},
+			options: {
+				name: 'options',
+				title: __( 'Options', 'woocommerce' ),
+			},
+		} ),
+		[ product.variations ]
+	);
 	return (
 		<>
-			{ formData && (
-				<>
-					<Sections sections={ formData.sections } />
-					<Fields fields={ formData.fields } />
-				</>
-			) }
+			<WooProductTabItem
+				id={ TAB_GENERAL_ID }
+				templates={ [ { name: TAB_GENERAL_ID, order: 1 } ] }
+				pluginId={ PLUGIN_ID }
+				tabProps={ tabPropData.general }
+			>
+				<WooProductSectionItem.Slot tab={ TAB_GENERAL_ID } />
+			</WooProductTabItem>
+			<WooProductTabItem
+				id={ TAB_PRICING_ID }
+				templates={ [ { name: TAB_GENERAL_ID, order: 3 } ] }
+				pluginId={ PLUGIN_ID }
+				tabProps={ tabPropData.pricing }
+			>
+				<WooProductSectionItem.Slot tab={ TAB_PRICING_ID } />
+			</WooProductTabItem>
+			<WooProductTabItem
+				id={ TAB_INVENTORY_ID }
+				templates={ [ { name: TAB_GENERAL_ID, order: 5 } ] }
+				pluginId={ PLUGIN_ID }
+				tabProps={ tabPropData.inventory }
+			>
+				<WooProductSectionItem.Slot tab={ TAB_INVENTORY_ID } />
+			</WooProductTabItem>
+			<WooProductTabItem
+				id={ TAB_SHIPPING_ID }
+				templates={ [ { name: TAB_GENERAL_ID, order: 7 } ] }
+				pluginId={ PLUGIN_ID }
+				tabProps={ tabPropData.shipping }
+			>
+				<WooProductSectionItem.Slot
+					tab={ TAB_SHIPPING_ID }
+					fillProps={ { product } }
+				/>
+			</WooProductTabItem>
+			{ window.wcAdminFeatures[ 'product-variation-management' ] ? (
+				<WooProductTabItem
+					id={ TAB_OPTIONS_ID }
+					templates={ [ { name: TAB_GENERAL_ID, order: 9 } ] }
+					pluginId={ PLUGIN_ID }
+					tabProps={ tabPropData.options }
+				>
+					<>
+						<OptionsSection />
+						<ProductVariationsSection />
+					</>
+				</WooProductTabItem>
+			) : null }
 		</>
 	);
 };
 
-/**
- * Preloading product form data, as product pages are waiting on this to be resolved.
- * The above Form component won't get rendered until the getProductForm is resolved.
- */
-resolveSelect( EXPERIMENTAL_PRODUCT_FORM_STORE_NAME ).getProductForm();
 registerPlugin( 'wc-admin-product-editor-form-fills', {
 	// @ts-expect-error 'scope' does exist. @types/wordpress__plugins is outdated.
 	scope: 'woocommerce-product-editor',
-	render: () => {
-		return <Form />;
-	},
+	render: () => (
+		<>
```

Comment:
We now just use the entry point files such as this one for the general product form, and the other for variants, to import the refactored section components and pass them the appropriate IDs.